### ISSUE TITLE
MissingAttribute "pads" in mnist-8.onnx

### ISSUE THREAD

**Issue by sid-alluri** [KarelPeeters/Kyanite#4]
Hi, Is there any end-to-end example for using kn-graph? I am trying to run inference for [mnist](https://github.com/onnx/models/blob/main/validated/vision/classification/mnist/model/mnist-8.onnx), and I am getting this error. 

> `thread 'main' panicked at src/kyan.rs:12:66:
> called `Result::unwrap()` on an `Err` value: MissingAttribute(Node { name: "Convolution28", op_type: "Conv" }, "pads", Ints, ["dilations", "strides", "auto_pad"])
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace`

My code looks as follows:
```
    let graph = load_graph_from_onnx_path("mnist-8.onnx", false).unwrap();

    let batch_size = 1;
    let inputs = [DTensor::F32(Tensor::zeros(IxDyn(&[batch_size, 1, 28, 28])))];

    let outputs: Vec<DTensor> = cpu_eval_graph(&graph, batch_size, &inputs);
```
thanks in advance for the help.

**Comment by KarelPeeters** [comment]
Hey! The main end-to-end example is [here](https://github.com/KarelPeeters/Kyanite?tab=readme-ov-file#quick-demo) in the readme, but you didn't miss anyhting, what you're doing is fully correct.

The error you're getting is because I didn't support the `auto_pad` attribute on convolutions yet. I've pushed a fix for that, but that exposed a missing operator implementation: `MaxPool`. I've implemented the basic cases for it, but sadly this network also uses the case where a dimension of size `14` is pooled by a kernel of size `3`, which is trickier to implement (since it doesn't divide equally). I'll fix this as soon as I find some more time.

More generally the repository you linked is a great resource, I should probably try to run all the networks in there through Kyanite to find any bugs and missing operators. I've added that as a todo to #2.

**Comment by sid-alluri** [comment]
Thank you for looking into it. I will give it a try.

**Comment by KarelPeeters** [comment]
I have a fix implemented for the max pooling as well, I'll test it in a couple of hours and make a new release containing both fixes.

**Comment by KarelPeeters** [comment]
This should be fixes as of fb9016fa1294c14bd69ca832d60ea000fc8a0aef. Let me know if it works and if you run into any more issues!

**Comment by KarelPeeters** [comment]
I'll assume this issue has been fixed, feel free to reopen or report a new one if it's not.
